### PR TITLE
Lift configfile

### PR DESCRIPTION
**Pull request by ewollesen**
Two smallish changes:

    Lift configfile.ConfigFile out of CredentialsStore
    
    Rather than assuming that users of CredentialStore will know to call Init
    after instantiation, the CredentialStore's dependency on a Docker *ConfigFile
    is made explicit by requiring it to be passed to the constructor. Now we need
    never worry that a user will forget to call Init, and it removes the awkward
    pattern of New-then-Init.
 

and:

    Don't store the RegistryPuller's StorageClient
    
    There's no point in storing it, as it's never reused.

### CODE REVIEW

**Comment by jonahjon** [comment]
/lgtm

Looks like a good refactor

File: pkg/registry/credentials.go [hunk 1]
```diff
@@ -1,44 +1,28 @@
...
-	"github.com/docker/cli/cli/config"
 	"github.com/docker/cli/cli/config/configfile"
 	"github.com/docker/cli/cli/config/credentials"
 	"oras.land/oras-go/v2/registry/remote/auth"
 )
 
-// CredentialStore for registry credentials such as ~/.docker/config.json.
-type CredentialStore struct {
-	directory  string
+// DockerCredentialStore for Docker registry credentials, like ~/.docker/config.json.
```
Comment (TerryHowe) [inline pkg/registry/credentials.go:9]: This is all fine, but I don't like docker in the name. I was trying to hide that. We don't have to be stuck with docker forever.

**Comment by vincentni** [comment]
We may want to revisit this PR as bundle pulling failed due to 401 unauthorized access.

**Pull request merged**